addons/website/models/res_users.py

```diff
@@ -72,10 +72,14 @@ def authenticate(cls, db, login, password, user_agent_env):
                 visitor_sudo = env['website.visitor']._get_visitor_from_request()
                 if visitor_sudo:
                     partner = env.user.partner_id
-                    partner_visitor = env['website.visitor'].sudo().search([('partner_id', '=', partner.id)])
+                    partner_visitor = env['website.visitor'].with_context(active_test=False).sudo().search([('partner_id', '=', partner.id)])
                     if partner_visitor and partner_visitor.id != visitor_sudo.id:
+                        # Link history to older Visitor and delete the newest
                         visitor_sudo.website_track_ids.write({'visitor_id': partner_visitor.id})
                         visitor_sudo.unlink()
+                        # Reactivate the older Visitor
```

Comment:
`# Reactivate the archived visitor (most likely archived by the cron for inactivity reasons)`